pkg/apis/core/v1/validation/validation_test.go

```diff
@@ -177,7 +178,7 @@ func TestValidateContainerResourceName(t *testing.T) {
 	}}
 	for _, tc := range successCase {
 		t.Run(tc.name, func(t *testing.T) {
-			if errs := ValidateContainerResourceName(tc.ResourceName, field.NewPath(tc.ResourceName)); len(errs) != 0 {
+			if errs := ValidateContainerResourceName(core.ResourceName(tc.ResourceName), field.NewPath(tc.ResourceName)); len(errs) != 0 {
```

Comment:
just define the struct field as `ResourceName` and these casts are not needed.  String literals are untyped and can be assigned to it without a cast